pkg/resources/validation/cluster/cluster.go

```diff
@@ -3,37 +3,65 @@ package cluster
 import (
 	"net/http"
 
+	"github.com/rancher/webhook/pkg/admission"
 	objectsv3 "github.com/rancher/webhook/pkg/generated/objects/management.cattle.io/v3"
-	"github.com/rancher/wrangler/pkg/webhook"
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
 	authenticationv1 "k8s.io/api/authentication/v1"
 	v1 "k8s.io/api/authorization/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	authorizationv1 "k8s.io/client-go/kubernetes/typed/authorization/v1"
 )
 
-func NewValidator(sar authorizationv1.SubjectAccessReviewInterface) webhook.Handler {
-	return &clusterValidator{
+var managementGVR = schema.GroupVersionResource{
+	Group:    "management.cattle.io",
+	Version:  "v3",
+	Resource: "clusters",
+}
+
+// NewValidator returns a new validator for management clusters.
+func NewValidator(sar authorizationv1.SubjectAccessReviewInterface) *Validator {
+	return &Validator{
 		sar: sar,
 	}
 }
 
-type clusterValidator struct {
+// Validator ValidatingWebhook for management clusters.
+type Validator struct {
 	sar authorizationv1.SubjectAccessReviewInterface
 }
 
-func (c *clusterValidator) Admit(response *webhook.Response, request *webhook.Request) error {
-	oldCluster, newCluster, err := objectsv3.ClusterOldAndNewFromRequest(request)
+// GVR returns the GroupVersionKind for this CRD.
+func (v *Validator) GVR() schema.GroupVersionResource {
+	return managementGVR
+}
+
+// Operations returns list of operations handled by this validator.
+func (v *Validator) Operations() []admissionregistrationv1.OperationType {
+	return []admissionregistrationv1.OperationType{admissionregistrationv1.Create, admissionregistrationv1.Update, admissionregistrationv1.Delete}
+}
+
+// ValidatingWebhook returns the ValidatingWebhook used for this CRD.
+func (v *Validator) ValidatingWebhook(clientConfig admissionregistrationv1.WebhookClientConfig) *admissionregistrationv1.ValidatingWebhook {
+	return admission.NewDefaultValidatingWebhook(v, clientConfig, admissionregistrationv1.ClusterScope)
```

Comment:
Does this need to set the failure policy to ignore?

Reply:
yes, good catch.